File: Assets/ProjectGame/Lib/data/class_data/PlaceData/BasePlaceData.js
import { BaseCustomClassData } from '../BaseCustomClassData.js';

export class BasePlaceData extends BaseCustomClassData {
    constructor() {
        super();
        this.position = [0.0, 0.0, 0.0]; // 座標
    }

    read(view, offset) {
        let o = offset;
        this.position = [view.getFloat32(o, true), view.getFloat32(o + 4, true), view.getFloat32(o + 8, true)]; o += 12;
        return o;
    }

    loadJson(data) {
        this.position = data.position ?? [0.0, 0.0, 0.0];
    }
}

File: Assets/ProjectGame/Lib/data/class_data/PlaceData/BasePlaceData.py
from ..BaseCustomClassData import BaseCustomClassData

class BasePlaceData(BaseCustomClassData):
    def __init__(self):
        super().__init__()
        self.position = [0.0, 0.0, 0.0]  # 座標

    def read(self, reader):
        self.position = [reader.read_float(), reader.read_float(), reader.read_float()]

    def load_json(self, data):
        self.position = data.get('position', [0.0, 0.0, 0.0])
